pyproject.toml

```diff
@@ -1,104 +1,76 @@
-furo = "^2023.9.10"
-myst-parser = "^2.0.0"
-readthedocs-sphinx-search = "^0.3.1"
-sphinx-copybutton = "^0.5.2"
-sphinx-togglebutton = "^0.3.2"
-sphinx_design = "^0.5.0"
-sphinx_inline_tabs = "^2023.4.21"
-mkdocs = "^1.5.2"
-mkdocstrings = {extras = ["python"], version = "^0.22.0"}
-mkdocs-material = "^9.1.21"
-mkdocs-section-index = "^0.3.5"
-mkdocs-autorefs = "^0.5.0"
-sphinx-autodoc2 = "^0.5.0"
-sphinx-book-theme = "^1.1.2"
-linkify-it-py = "^2.0.3"
-
-
-[tool.poetry.group.dev.dependencies]
-ipykernel = "^6.26.0"
-pytest = "^7.4.3"
-pytest-cov = "^4.1.0"
-mypy = "^1.6.1"
-black = "^23.10.1"
-isort = "^5.12.0"
-ruff = "^0.3.0"
-
-[tool.poetry.group.test.dependencies]
-mypy = "^1.6.1"
-pytest = "^7.4.3"
-pytest-cov = "^4.1.0"
-pytest-mock = "^3.12.0"
-
-[tool.poetry-dynamic-versioning]
-enable = false
-bump  = false
-vcs = "git"
-format = "v{base}{stage}{revision}"
-
-
-[tool.poetry-dynamic-versioning.substitution]
-files = ["dhlab/_version.py"]
-
-[build-system]
-requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
-build-backend = "poetry_dynamic_versioning.backend"
+dependencies = [
+    "ipython>=8.17.2",
+    "matplotlib>=3.8.1",
+    "networkx>=3.2.1",
+    "pandas>=2.2.3",
+    "python-louvain>=0.16",
+    "requests>=2.31.0",
+    "seaborn>=0.13.2",
+    "scipy==1.14",
+    "openpyxl>=3.1.2",
+    "beautifulsoup4>=4.12.2",
+    "jinja2>=3.1.4",
+    "tqdm>=4.66.6",
+    "numpy==1.26",
+]
 
-[tool.isort]
-profile = "black"
+[project.urls]
+Homepage = "https://www.nb.no/dh-lab/"
+Documentation = "https://nationallibraryofnorway.github.io/DHLAB/"
+Repository = "https://github.com/NationalLibraryOfNorway/DHLAB"
+Issues = "https://github.com/NationalLibraryOfNorway/DHLAB/issues"
+Tutorials = "https://nationallibraryofnorway.github.io/digital_tekstanalyse/tutorial.html"
 
-[tool.mypy]
-files = "tests"
-mypy_path = "dhlab"
-namespace_packages = true
-explicit_package_bases = false
-show_error_codes = true
-warn_unused_configs = true
-strict = true
-enable_error_code = [
-    "ignore-without-code",
-    "redundant-expr",
-    "truthy-bool",
+[project.optional-dependencies]
+docs = [
+    "Sphinx>=7.2.6",
+    "furo>=2023.9.10",
+    "myst-parser>=2.0.0",
+    "readthedocs-sphinx-search>=0.3.1",
+    "sphinx-copybutton>=0.5.2",
+    "sphinx-togglebutton>=0.3.2",
+    "sphinx_design>=0.5.0",
+    "sphinx_inline_tabs>=2023.4.21",
+    "mkdocs>=1.5.2",
+    "mkdocstrings[python]>=0.22.0",
+    "mkdocs-material>=9.1.21",
+    "mkdocs-section-index>=0.3.5",
+    "mkdocs-autorefs>=0.5.0",
+    "sphinx-autodoc2>=0.5.0",
+    "sphinx-book-theme>=1.1.2",
+    "linkify-it-py>=2.0.3",
+]
+dev = [
+    "ipykernel>=6.26.0",
+    "pytest>=7.4.3",
+    "pytest-cov>=4.1.0",
+    "mypy>=1.6.1",
+    "black>=23.10.1",
+    "isort>=5.12.0",
+    "ruff>=0.3.0",
+]
+test = [
+    "mypy>=1.6.1",
+    "pytest>=7.4.3",
+    "pytest-cov>=4.1.0",
+    "pytest-mock>=3.12.0",
 ]
 
+[build-system]
+requires = ["pdm-backend"]
+build-backend = "pdm.backend"
```

Comment:
We should either 
1. use a `src`-layout, placing the `dhlab`-directory inside a `src`-directory,
2. or specify that we have a flat layout for the build backend (see the pdm documentation [here](https://backend.pdm-project.org/build_config/#include-or-exclude-files)).

I'm leaning towards 1. since it would make it easier for tests to pick up on installation errors. This also seems to be the prefered layout these days (see e.g. the [tutorial on packaging.python.org](https://packaging.python.org/en/latest/tutorials/packaging-projects/))

Reply:
Good suggestion!